frame/evm/precompile/secp256r1/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+// This file is part of Frontier.
+
+// Copyright (c) Moonsong Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// “secp256r1” is a specific elliptic curve, also known as “P-256” and “prime256v1” curves.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
+
+use p256::ecdsa::{signature::hazmat::PrehashVerifier, Signature, VerifyingKey};
+use fp_evm::{ExitSucceed, PrecompileResult, Precompile, PrecompileHandle, PrecompileOutput};
+
+pub struct P256Verify;
+
+impl P256Verify {
+	/// https://github.com/ethereum/RIPs/blob/master/RIPS/rip-7212.md#precompiled-contract-gas-usage
+	const BASE_GAS: u64 = 3_450;
```

Comment:
We should not use the base gas from the EIP as it was designed for a native execution.
We should add a benchmark scenario on the precompile-benchmark pallet to know how many weight to charge and charge with `record_external_cost`

Reply:
Fixed on moonbeam side